colossalai/booster/plugin/hybrid_parallel_plugin.py

```diff
@@ -456,42 +457,33 @@ def configure(
-                        optimizer.working_to_master_map, optimizer.master_to_working_map
-                    )
+                if self.precision in ['fp16', 'bf16']:
+                    optimizer = HybridParallelAMPOptimizer(optimizer,
+                                                           model,
+                                                           use_pipeline=self.enable_pipeline_parallelism,
+                                                           param_info=param_info,
+                                                           precision=self.precision,
+                                                           max_norm=self.max_norm,
+                                                           **self.amp_config)
                 else:
                     optimizer = HybridParallelNaiveOptimizer(
                         optimizer, model, use_pipeline=self.enable_pipeline_parallelism, param_info=param_info
                     )
             else:
                 assert self.dp_size > 1, "Please use Zero when data parallel size is greater than 1."
-                assert self.precision != "fp32", "Please set precision to 'fp16' or 'bf16' when using ZeRO."
-                optimizer = HybridParallelZeroOptimizer(
-                    optimizer,
-                    model,
-                    use_pipeline=self.enable_pipeline_parallelism,
-                    param_info=param_info,
-                    dp_process_group=self.dp_group,
-                    tp_process_group=self.tp_group,
-                    verbose=True,
-                    clip_grad_norm=self.max_norm,
-                    **self.zero_config,
-                    **self.amp_config,
-                )
-                self.checkpoint_io.link_master_and_working_param(
-                    optimizer._param_store.working_to_master_param, optimizer._param_store.master_to_working_param
-                )
-
+                assert self.precision != 'fp32', "Please set precision to 'fp16' or 'bf16' when using ZeRO."
+                optimizer = HybridParallelZeroOptimizer(optimizer,
+                                                        model,
+                                                        use_pipeline=self.enable_pipeline_parallelism,
+                                                        param_info=param_info,
+                                                        dp_process_group=self.dp_group,
+                                                        tp_process_group=self.tp_group,
+                                                        verbose=True,
+                                                        clip_grad_norm=self.max_norm,
+                                                        **self.zero_config,
+                                                        **self.amp_config)
+        # inject update_master_params
+        model.module.update_master_params = MethodType(optimizer.update_master_params, model)
```

Comment:
Only inject when optimizer is not None. Maybe you should add a tab

Reply:
modified